src/test_trie.py

```diff
@@ -0,0 +1,196 @@
+    init_empty_trie.insert('morgan')
+    init_empty_trie.insert('borgan')
+    assert init_empty_trie.contains('morgan') is True
+    assert init_empty_trie.contains('borgan') is True
+    assert init_empty_trie._root._children[0]._data == 'm'
+    assert init_empty_trie._root._children[1]._data == 'b'
+    assert init_empty_trie._root._children[0]._children is not init_empty_trie._root._children[1]._children
+    assert len(init_empty_trie._root._children) == 2
+
+
+def test_contains(init_trie_one_word):
+    """Check error is raised when iterable inserted."""
+    assert init_trie_one_word.contains('cake') is True
+    assert init_trie_one_word.contains('ca') is False
+
+
+def test_contains_invalid_input_int(init_trie_one_word):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(TypeError):
+        init_trie_one_word.contains(3)
+
+
+def test_contains_invalid_input_iterable(init_trie_one_word):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(TypeError):
+        init_trie_one_word.contains(['cake'])
+
+
+def test_remove_word_empty_trie(init_empty_trie):
+    """Test remove on empty trie."""
+    with pytest.raises(ValueError):
+        init_empty_trie.remove('you')
+
+
+def test_remove_float_trie(init_trie_one_word):
+    """Test remove on empty trie."""
+    with pytest.raises(TypeError):
+        init_trie_one_word.remove(1.0)
+
+
+def test_remove_non_existent_word_trie(init_trie_one_word):
+    """Test remove on empty trie."""
+    with pytest.raises(ValueError):
+        init_trie_one_word.remove('you')
+
+
+def test_remove_words_shared_beginning(init_trie_three_words_shared_letters):
+    """Test remove on words with shared beginning letters."""
+    init_trie_three_words_shared_letters.remove('be')
+    assert init_trie_three_words_shared_letters.contains('be') is False
+    init_trie_three_words_shared_letters.remove('bell')
+    assert init_trie_three_words_shared_letters.contains('bell') is False
+    assert init_trie_three_words_shared_letters.contains('bee') is True
+
+
+def test_size_trie_before_and_after_remove(init_trie_three_words_shared_letters):
+    """Test trie size before and after a removal."""
+    assert init_trie_three_words_shared_letters.size() == 3
+    init_trie_three_words_shared_letters.remove('bee')
+    assert init_trie_three_words_shared_letters.size() == 2
+
+
+def test_size_trie_before_and_after_insert(init_trie_three_words_shared_letters):
+    """Test trie size before and after a removal."""
+    assert init_trie_three_words_shared_letters.size() == 3
+    init_trie_three_words_shared_letters.insert('happy')
+    assert init_trie_three_words_shared_letters.size() == 4
+
+
+def test_insert_jumbo_list(init_jumbo_list):
+    """."""
+    jumbo_trie = Trie()
+    for word in init_jumbo_list:
+        jumbo_trie.insert(word)
+    assert len(init_jumbo_list) == jumbo_trie.size()
+    assert jumbo_trie.contains('apple')
```

Comment:
I would also write a test for remove that uses the large trie. Also test that if you remove a larger word like 'apples' the word 'apple' still exists.